examples/policies/auth/auth.yaml

```
@@ -0,0 +1,14 @@
+apiVersion: "cilium.io/v2"
+kind: CiliumNetworkPolicy
+metadata:
+  name: "l3-egress-rule"
+spec:
+  endpointSelector:
+    matchLabels:
+      run: ssh-client
+  egress:
+  - toEndpoints:
+    - matchLabels:
+        run: sshd
+    auth:
+      type: test
```

I think that making this new field a structure is a good idea, but I don't know what `type: test` is supposed to indicate. Perhaps..?

```suggestion
      enabled: true
```

I'm assuming that the primary intended mechanism for choosing the auth plugin module will be through agent configuration, not policy-by-policy so hopefully we don't need a type here. Unless we already have some idea of the different types of authentication we plan to support.

---

`type: test` is a stand-in for a real auth method that allows verifying the functionality in this PR. Not sure if renaming this to something like `type: none` would help & can we ship CIlium with this in? `none` would do the roundtrips between the kernel and userspace exercising the new functionality without actually performing any authentication.

I'll change `type: test` to `type: none` in the meanwhile.

---

I like `type: "null"`, it's a little odd (looking like a test example) but communicates pretty well the kind of authentication that is really going on here.